android/src/main/java/com/cuhacking/atlas/MainActivity.kt

```diff
@@ -59,26 +61,30 @@ class MainActivity : AppCompatActivity() {
         }
         binding.searchResultsList.adapter = searchResultsAdapter
 
-        val searchResultsObserver = Observer<List<SearchResult>> { searchResultItems ->
-            searchResultsAdapter.submitList(searchResultItems)
+        lifecycleScope.launch {
+            searchViewModel.searchResults.collect { searchResultItems ->
+                searchResultsAdapter.submitList(searchResultItems)
+            }
+        }
+            binding.searchView.setOnQueryTextListener(object :
+                    SearchView.OnQueryTextListener {
+
+                override fun onQueryTextChange(newText: String): Boolean {
+                    lifecycleScope.launch {
+                    searchViewModel.getSearchResults(newText.trim())
+                    }
+                    binding.searchResultsList.visibility = View.VISIBLE
```

Comment:
```suggestion
                    binding.searchResultsList.isVisible = true
```

I think(?) we have android kotlin extension functions included? If not, we could add them: https://developer.android.com/kotlin/ktx (core-ktx).